### PR TITLE
Add support for Kafka 3.5.1 and 3.6.0

### PR DESCRIPTION
**Pull request by see-quick**
This PR adds Kafka 3.5.1 and 3.6.0-rc0 versions. Moreover, it also changes a script, which downloads Kafka binaries. It adds another condition if kafka version is not even in the archive repository i.e. `https://archive.apache.org/dist/kafka/` it starts downloading from `https://home.apache.org/~satishd`.

Also removes 3.5.0 as we only support the latest from the minor releases.

### CODE REVIEW

File: images/download_kafka.sh [hunk 1]
```diff
@@ -20,20 +20,34 @@ SCALA_VERSION=$(cat supported_scala.version)
...
-      wget $KAFKA_URL -P "$KAFKA_DOWNLOADED_TARS_DIR"
+        echo "There was a problem downloading Kafka from $KAFKA_URL, we are gonna download it from archives."
+        KAFKA_URL="https://archive.apache.org/dist/kafka/$KAFKA_VERSION/kafka_$SCALA_VERSION-$KAFKA_VERSION.tgz"
+
+        if wget $KAFKA_URL -P "$KAFKA_DOWNLOADED_TARS_DIR"
+        then
+            echo "We successfully download Kafka from $KAFKA_URL"
+        else
+            echo "There was a problem downloading Kafka from $KAFKA_URL, we are gonna download it from release candidate."
+            KAFKA_URL="https://home.apache.org/~satishd/kafka-$KAFKA_VERSION-rc0/kafka_2.13-$KAFKA_VERSION.tgz"
```
Comment (scholzj) [inline images/download_kafka.sh:41]: This address will differ for every release. So I do not think this should be hardcoded in the code.

**Comment by see-quick** [comment]
> I'm not sure what the goal is of this PR. We normally open these only as drafts for testing purposes with RCs and merge them only with GAs.

I wanted to have a 3.6.0-rc0 image in the main branch. Therefore I could use them and verify them in the `test-containers` repository.  And when Kafka 3.6.0 is released I just replace the shas. I can also wait for the 3.6.0 to be released but wanted to try this way to be as quick as possible.

**Comment by scholzj** [comment]
Well, if nothing else, I think the URL hardcoded in the script is a terrible hack. I do not think that should be merged into main branch. You would need to solve that in a better way. You should also not present RC version of 3.6.0 as 3.6.0. You would need to add it a `3.6.0-rc0` or something like that to keep the versioning clean.

**Comment by see-quick** [comment]
> Well, if nothing else, I think the URL hardcoded in the script is a terrible hack. I do not think that should be merged into main branch. You would need to solve that in a better way. You should also not present RC version of 3.6.0 as 3.6.0. You would need to add it a 3.6.0-rc0 or something like that to keep the versioning clean.

Okay, I think that's too much effort to do so only for `rc0` testing (it's easier to do so locally everything and avoid this changes). I will wait for 3.6.0 GA then and (i.) remove a code for downloading the release candidate, and (ii.) update shas for 3.6.0. Thanks for your input :)

**Review by Frawless — APPROVED** [review]

**Review by scholzj — APPROVED** [review]

**Pull request merged**